src/porepy/numerics/time_step_control.py

```diff
@@ -1,130 +1,191 @@
-from typing import Tuple, Union
+from typing import Optional, Tuple, Union
+
+import numpy as np
 
 __all__ = ["TimeSteppingControl"]
 
 
 class TimeSteppingControl:
-    """Parent class for iteration-based time stepping control routine."""
+    """Parent class for iteration-based time stepping control routine.
+
+    Parameters:
+        schedule: List containing the target times for the simulation.
```

Comment:
We can imagine combining this with a scheduler which determines necessary target times due to visualization, changes in well rates or boundary conditions etc. No need to adjust anything here, but let's keep this in mind.

Reply:
I agree, this is will be a good enhancement. However, I believe the PR is already too long and I suggest dealing with this feature request in a different PR. Is that OK with you?

Reply:
It is! Should we make an issue?

Reply:
Sure, I can create one.